torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,448 @@
+class CallgrindStats(object):
+    stmt: str
+    setup: str
+    number: int
+    num_threads: int
+    built_with_debug_symbols: bool
+    baseline_inclusive_stats: Tuple[Tuple[int, str], ...]
+    baseline_exclusive_stats: Tuple[Tuple[int, str], ...]
+    stmt_inclusive_stats: Tuple[Tuple[int, str], ...]
+    stmt_exclusive_stats: Tuple[Tuple[int, str], ...]
+
+    def __repr__(self) -> str:
+        newline = "\n"  # `\` cannot appear in fstring code section.
+        base_stats = self.baseline_exclusive_stats
+        self_stats = self.stmt_exclusive_stats
+        output = textwrap.dedent(f"""
+        {super().__repr__()}
+          stmt:  {self.stmt.replace(newline, newline + ' ' * 9)}
+          setup: {self.setup.replace(newline, newline + ' ' * 9)}
+          {self.num_threads} thread{'s' if self.num_threads > 1 else ''}
+        {'':>25}All{'':>10}Noisy symbols removed
+          Instructions: {self._counts(self_stats, True):>12}{'':>15}{self._counts(self_stats, False):>12}
+          Baseline:     {self._counts(base_stats, True):>12}{'':>15}{self._counts(base_stats, False):>12}
+        """).strip()
+        if not self.built_with_debug_symbols:
+            output += textwrap.dedent("""
+            Warning: PyTorch was not built with debug symbols.
+                     Source information may be limited. Rebuild with
+                     REL_WITH_DEB_INFO=1 for more detailed results.""")
+        return output
+
+    def stats(self, inclusive: bool = False) -> Tuple[Tuple[int, str], ...]:
+        """Returns stats as a tuple of (count, function)
+
+        `inclusive` matches the semantics of callgrind. If True, the counts
+        include instructions executed by children. `inclusive=True` is useful
+        for identifying hot spots in code; `inclusive=False` is useful for
+        identifying reducing noise when diffing counts from two different
+        runs. (See CallgrindStats.delta(...) for more details)
+        """
+        if inclusive:
+            first, second = self.stmt_inclusive_stats, self.baseline_inclusive_stats
+        else:
+            first, second = self.stmt_exclusive_stats, self.baseline_exclusive_stats
+        return self._diff(first, second)
+
+    def counts(self, include_lookdict_unicode: bool = True) -> int:
+        """Returns the total number of instructions executed.
+
+        Several instructions in the CPython interpreter are rather noisy. These
+        instructions involve unicode to dictionary lookups which Python uses to
+        map variable names. By default these are included, but setting
+        `include_lookdict_unicode=False` will exclude them and generally lead
+        to less noisy counts.
+        """
+        return self._counts(self.stmt_exclusive_stats, include_lookdict_unicode)
+
+    # FIXME: Once 3.7 is the minimum version, type annotate `other` per PEP 563
+    def delta(self, other, inclusive: bool = False, subtract_baselines: bool = True):
```

Comment:
Missing return type.

Reply:
Also, I personally prefer "Before - After" nomenclature for diffs, makes it clear which direction the delta goes.

Reply:
...which would have been useful for me right now, because in fact self is after, and other is before, isn't it? As I was typing that previous comment, I had them swapped.

Reply:
It's funny; my formal training is in chemistry where delta = "final - initial" was ground into my skull. In this case `delta` seems analogous to `__sub__` which I think matches this convention?